dask/array/core.py

```diff
@@ -2852,10 +2838,25 @@ def choose(a, choices):
 """.strip()
 
 
+@wraps(np.argwhere)
+def argwhere(a):
+    from .creation import indices
+
+    nz = (a != 0).flatten()
+
+    ind = indices(a.shape, dtype=np.int64, chunks=a.chunks)
+    ind = ind.reshape((a.ndim, -1)).T
+    ind = compress(nz, ind, axis=0)
+
+    return ind
+
+
 @wraps(np.where)
 def where(condition, x=None, y=None):
-    if x is None or y is None:
-        raise TypeError(where_error_message)
+    if (x is None) != (y is None):
+        raise ValueError("either both or neither of x and y should be given")
```

Comment:
Made this a `ValueError` to match NumPy's behavior. Though it is worth noting it was a `TypeError` before.